### ISSUE TITLE
ufunc specifications

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas2#68]
looks ``scipy`` encapsulated lots of function type things here: https://github.com/scipy/scipy/pull/6509

we eventually will want to enable things like this as well for ufunc like things that can be processed efficiently via the c++ code.

**Comment by wesm** [comment]
This is cool. I think we'll want to have a similar low level callable interface for functions that utilize pandas2's C++ API

**Comment by jreback** [comment]
yep. arguably this interface should exist even lower down the chain (e.g. in ``pyarrow``) or utilities. maybe even get ``scipy`` to use it :>

**Comment by wesm** [comment]
Yeah, definitely. Having a native callable API for Arrow data in C++ (that writes output into preallocated memory, or uses one of the builder classes, e.g. for strings) would be very nice
